storagenode/pieces/cache_test.go

```diff
@@ -332,3 +332,21 @@ func TestCacheCreateMultipleSatellites(t *testing.T) {
 	})
 
 }
+
+func TestConcurrency(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 0,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		node := planet.StorageNodes[0]
+		satellite := planet.Satellites[0]
+
+		done := make(chan bool)
```

Comment:
Use  `ctx.Go` (or `errgroup.Group`) rather than channels and `go func`. They are easy to get wrong. While the current code is correct, I'd prefer if we didn't use `go func` directly, unless exceptional cases.